src/dialects/abstract/abstract-query-interface.ts

```diff
@@ -0,0 +1,23 @@
+import type { Sequelize } from 'src/sequelize';
+import type { AbstractQueryGenerator } from './query-generator';
+import type { QueryInterface } from './query-interface';
+
+class AbstractQueryInterface implements QueryInterface {
```

Comment:
We don't use default exports in the project to maximise compat between esm & cjs

```suggestion
export class AbstractQueryInterface implements QueryInterface {
```

Reply:
Gotcha - what's the difference between this TS file and `query-interface.d.ts`? Would `query-interface.d.ts` contain the supporting types and `AbstractQueryInterfaceTypeScript` would contain the actual interface definition?